### ISSUE TITLE
rowid does not exist in Yugabyte dialect delete

### ISSUE THREAD

**Issue by wdroste** [jOOQ/jOOQ#17531]
### Expected behavior

Expect this query to succeed.

```
  public void cleanup(@NonNull Duration olderThan) {
    final var ts = Instant.now().minus(olderThan).toEpochMilli();
    final var olderThanTs = new Timestamp(ts);
    ctx.transaction(
        tx ->
            ctx.deleteFrom(Schema.MGR)
                .where(Columns.END_DATE.lessOrEqual(olderThanTs))
                .orderBy(Columns.END_DATE.asc())
                .limit(1000L)
                .execute());
  }
```

### Actual behavior

Throws an exception:

```
SQL [delete from "brq_queue_manager" where "brq_queue_manager".rowid in (select "brq_queue_manager".rowid from "brq_queue_manager" where end_date <= cast(? as timestamp) order by end_date asc limit ?)]; ERROR: column brq_queue_manager.rowid does not exist
  Position: 76
```

### Steps to reproduce the problem

Simple table w/ for columns and do a delete w/ a limit.

### jOOQ Version

JOOQ Professional 3.19.11

### Database product and version

Yugabyte 2.14.4.0-b26

### Java Version

JDK 11

### JDBC / R2DBC driver name and version (include name if unofficial driver)

pgjdbc v42.7.3-yb-1

**Comment by lukaseder** [comment]
Thanks for your report. YugabyteDB doesn't support any kind of rowid yet, including PostgreSQL's `ctid` column. Try:

```sql
SELECT ctid FROM t
```

Producing:

> ERROR: System column "ctid" is not supported yet

Without such a feature, it seems there's no way of emulating `DELETE .. ORDER BY .. LIMIT` unless you add a primary key, of course.

**Unrelated**

You should use the transactional `Configuration` in your lambda, not the outer `DSLContext`:

```java
ctx.transaction(
        tx ->
            tx.dsl().delete(...) // Use tx here, not ctx
```

**Comment by wdroste** [comment]
I'm a little puzzled by this since the dialect is Yugabyte why can't jOOQ work w/ this query as its pretty common and given the context available couldn't jOOQ work around this. I finally just used a subquery to feed the delete `DELETE` .. and this was closed w/o even noting that jOOQ didn't at least give me a `NotSupported` query it instead tried something you obviously know it can't do..

**Comment by lukaseder** [comment]
> I'm a little puzzled by this since the dialect is Yugabyte why can't jOOQ work w/ this query as its pretty common and given the context available couldn't jOOQ work around this. I finally just used a subquery to feed the delete `DELETE` ..

Is anything unclear about my previous explanation? Let me try again:

- YugabyteDB doesn't support `ORDER BY .. LIMIT` natively
- The standard emulation uses a self-join on the primary key, if present, or the `ROWID` if no primary key is present (though obviously, the concept of a `ROWID` needs to be supported for this emulation to work)
- I currently don't see any other way to emulate this

> Its a bit disappointing this was "Closed" the reason being Yugabyte doesn't support it.

What would an alternative resolution be? There doesn't seem to be anything to do here, right now. Future evolutions might include:

- YugabyteDB starts supporting `ORDER BY .. LIMIT` (in case of which, a different issue will be created to track this)
- YugabyteDB starts supporting `ROWID` (in case of which, again, a different issue will be created, as this doesn't have anything to do with `DELETE` per se)
- A new way to emulate the statement is discovered, in case of which a different issue will be created

> jOOQ created the `not supported` query so how is this Yugabyte's issue? At least throw `Not supported` vs attempting something that obviously is _not_ supported.

We could try to throw such exceptions all over the place. In fact, we:

- Offer a checker / error prone rule to validate your own API usage: https://www.jooq.org/doc/latest/manual/tools/checker-framework. This doesn't help in this case, because jOOQ internals call the unsupported API, not client code.
- Have an internal test utility that instruments all of our API to make sure we don't run tests against unsupported API (leading to false positives). In this case, the test simply wasn't run, because jOOQ internals call `Table.rowid()`, which isn't supported by YugabyteDB

I think there's an existing issue for this improvement. It would just be a different exception though, other than the one thrown by the database dialect already, so it's mostly a cosmetic improvement.

**Comment by wdroste** [comment]
As cosmetic as it seems a proper exception would have directed me to avoid opening this ticket. As it can provide at least its not supported and possibly a suggestion. It would have saved me significant time.

**Comment by lukaseder** [comment]
@wdroste Sure, I can see that this is annoying. I didn't mean for the "cosmetic improvement" to imply that this isn't important. I just wanted to stress that this doesn't appear to be something jOOQ can work around.

**Comment by lukaseder** [comment]
I found the issue that's requesting the same thing as you are:

- https://github.com/jOOQ/jOOQ/issues/9319

If we implement this, we can still also implement a `Settings` that decides whether we'll just render a comment (e.g. ideal for the translator use-case), or throw an exception.
